tools/ports/sdl2_image.py

```diff
@@ -52,6 +52,11 @@ def create(final):
     if 'jpg' in settings.SDL2_IMAGE_FORMATS:
       defs += ['-sUSE_LIBJPEG']
 
+    if settings.USE_PTHREADS:
+      defs += ['-sUSE_PTHREADS']
+    if settings.WASM_WORKERS:
+      defs += ['-sWASM_WORKERS']
```

Comment:
Doesn't this file also need to declare a variant for each of pthreads and wasm-workers, like the others? Or how does that work?

Reply:
This one probably isn't needed...  since a pthread variant was not previously needed.

Reply:
My understanding:
The variant on each file is mainly used by the embuilder.py to clean / rebuild the target
The resulting library name is determined by get_lib_name

Because sdl2_image depends on sdl2, sdl2 have a variant libsdl2-mt.a
The settings here is also used by the dependenices
Therefore, we need to defs += ['-sUSE_PTHREADS'] so that it look for "libsdl2-mt.a" instead of "libsdl2.a"
Otherwise, some error message will popup like: "failed to lock on libsdl2.a"

Reply:
But until now we have not needed a `-mt` flavor of sdl2_image.   Presumably folks are using sdl2-mt without issue ... can you give and example of something that fails today using sdl2_image and pthreads?  
If you are correct this should be failing today using pthreads.

Reply:
Sorry I missed this comment, I am just trying to build with `-s USE_PTHREADS=1 -s USE_SDL=2 -s USE_SDL_IMAGE=2 -s SDL2_IMAGE_FORMATS=`

For example:

```
emcc -g -s USE_PTHREADS=1 -s USE_SDL=2 -s USE_SDL_IMAGE=2 -s SDL2_IMAGE_FORMATS="[""png""]" --preload-file ../resources@/ -o main.html main.cpp
```

But I will get the error: AssertionError: attempt to lock the cache while a parent process is holding the lock (sysroot\lib\wasm32-emscripten\libSDL2.a)

This error doesn't make sense to me, because only libSDL2-mt.a is found

![image](https://user-images.githubusercontent.com/1687930/210293446-6448c9d3-bfb1-4a0d-a9dc-21e6c6bae885.png)


![image](https://user-images.githubusercontent.com/1687930/210293232-04a5f9ac-930c-4110-b395-af0ff339e0f5.png)